static/app/views/starfish/views/spanSummary/queries.tsx

```diff
@@ -5,14 +5,39 @@ export const getSpanSamplesQuery = ({
   groupId,
   transactionName,
   user,
+  sortBy,
   datetime,
+  p50,
 }: {
   groupId;
   transactionName;
   user;
   datetime?: DateTimeObject;
+  p50?: number;
+  sortBy?: string;
 }) => {
   const {start_timestamp, end_timestamp} = datetimeToClickhouseFilterTimestamps(datetime);
+
+  if (sortBy === 'median_samples') {
+    return `
+      SELECT transaction_id, transaction, description, user, domain, span_id, sum(exclusive_time) as exclusive_time, abs(minus(exclusive_time, ${p50})) as diff
+        FROM spans_experimental_starfish
+        WHERE group_id = '${groupId}'
+        ${transactionName ? `AND transaction = '${transactionName}'` : ''}
+        ${user ? `AND user = '${user}'` : ''}
+        ${
+          start_timestamp
+            ? `AND greaterOrEquals(start_timestamp, '${start_timestamp}')`
+            : ''
+        }
+        ${end_timestamp ? `AND lessOrEquals(start_timestamp, '${end_timestamp}')` : ''}
+        GROUP BY transaction_id, transaction, description, user, domain, span_id
+        HAVING lessOrEquals(divide(diff, ${p50}), 0.05)
```

Comment:
I think for sparse samples this might be too strict, 0.05 is a pretty narrow window! Sorting by `diff` should be good enough, no? Also, if you get rid of the `HAVING` you won't need to copypasta the whole query, you can add the `as diff` part, and have the `ORDER BY` field change based on `sortBy` 😉

Reply:
Ah yes, great points. I guess my thinking was that if data was too sparse, we would see weird results if we selected the median option. But that's not going to be a problem in the real thing, so your suggestion makes the most sense!

Reply:
Up to you if you want to bother fixing this right now, but worth thinking about!

Reply:
I'm gonna fix this in the next PR I make (related to our discussion on Slack today)